client/homebrew/pages/editPage/editPage.jsx

```diff
@@ -141,7 +166,15 @@ const EditPage = createClass({
 	trySave : function(){
 		if(!this.debounceSave) this.debounceSave = _.debounce(this.save, SAVE_TIMEOUT);
 		if(this.hasChanges()){
-			this.debounceSave();
+			if(this.isEdit()) {
+				this.debounceSave();
+			}
+			if(this.isNew()) {
+				//console.log(this.state.brew.text);
+				localStorage.setItem(BREWKEY, this.state.brew.text);
+				localStorage.setItem(STYLEKEY, this.state.brew.style);
+				this.debounceSave.cancel();
+			}
```

Comment:
It can be moved; `save()` is called by the Save button click handler but also this automatic save function, so those two states will need to be disambiguated.

Reply:
`trySave()` should trigger automatically when a user types (or edits metadata, or changes google Drive status) and we detect that the content has changed. This just sets a timer for `SAVE_TIMEOUT` miliseconds and ignores any other save requests until that time is up, at which point it calls `save()`. If the user undoes their change before that happens, the save will not trigger if I remember right.

We can keep the save button calling `save()` directly since I don't think users want to click and nothing happens for 3 seconds.

Reply:
Yes, absolutely; what I meant is that because `save()` is called by clicking the SAVE button, it causes the Brew to be saved to the appropriate HB/Google storage and redirects to the `/edit` page, whereas this automatic saving from `/new` only saves to localStorage. So if we put it in the `save()`, we'll need a way to identify that this should be saved to localStorage, not MongoDB or Google Drive.

We could also create a new function, e.g. `saveLocal()`, and drop the three lines of code into that.

Reply:
I'm now wondering about a `this.saveLocation` variable with possible values of `Homebrewery`, `GoogleDrive`, `BrowserCache`, which direct the `save()` logic. It would be more extensible than just using the single `this.saveGoogle` boolean.

Probably outside the scope of this PR, though. Maybe in a refactor PR.

Reply:
Ah I understand now. Yes that sounds like a good approach. I don't mind if you want to add that in so you can move the local storage saving to the save function.

Reply:
Although you are right it might not be that straightforward.

Reply:
Maybe if saving comes from TrySave, pass a "autosave" parameter to the Save function and then we can handle it in there depending if it's true or false.